buffer/llbuffer.go

```diff
@@ -0,0 +1,107 @@
+package buffer
+
+import (
+	"container/list"
+	"io"
+	"sync"
+)
+
+type FileChunk struct {
+	Start   int64
+	Content []byte
+}
+
+type OrderedBuffer struct {
+	w             io.Writer
+	totalExpected int64
+	written       int64
+	list          *list.List
+	mu            *sync.Mutex
+	Queued        int
```

Comment:
Does `Queued` need to be exported?

Reply:
It was a statistic, I think removing it completely would be better. The tests use it as an assertion that no items are left in the queue, but the assertion that checks the output would fail first if any items were left in the buffer. Just checking the result is sufficient, hence I will remove the field completely.